scripts/nbconverted/0_process_data.py

```diff
@@ -0,0 +1,284 @@
+# 3. Only keep PA genes, remove control genes
+# 
+# ```
+# normalized_data
+# ```
+# 1. Use data from above
+# 2. Normalize each gene to be between 0 and 1
+
+# ## About Affymetrix GeneChip processing
+# 
+# **Measurements**
+# mRNA samples samples are labeled with flouresence and hybridized to GeneChip probe array.  The probe array is then scanned and the flouresence intensity of each probe (or feature) is measured.  A trasncript is represented by a probe set (~11-20 pairs of probes - see explanation of pairs below).  The probe set intensity forms the expression measure for a given transcript.
+# 
+# **Array Design**
+# Two probes: 1) probe is completely complementary to target sequence, perfect match probe (PM) and 2) probe contains a single mismatch to the target sequence in the middle of the probe, mismatch probe(MM).  A probe pair is (PM, MM)  
+# 
+# from [The Affymetrix GeneChip Platform: An Overview](https://www.sciencedirect.com/science/article/pii/S0076687906100014?via%3Dihub#fig0001)
+# 
+# **Robust multiarray average (rma)**
+# 
+# 1. Assuming PM = background + signal we want to correct for background signal, returns E[signal|background+signal] assuming signal~exponential and background~normal.
+# 2. Use quantile normalization is to make the distribution of probe intensities the same across arrays.  The steps are 1) for each array, rank the probe intensity from lowest to highest, 2) For each array rearrange probe intensity values from lowest to highest, 3) Take the average across arrays for each probe and asssign rank, 4) replace ranks from (1) with mean values.  Example from [Quantile Normalization wiki](https://en.wikipedia.org/wiki/Quantile_normalization)   
+# 3. Calculating the probe set intensity by averaging PM-MM across probes in probe set and log2 transform, Y.  Fit regression model to Y (probe set intensity) = probe affinity effect + *log scale expression level* + error
+# 
+# from [Exploration, normalization, and summaries of high density oligonucleotide array probe level data](https://academic.oup.com/biostatistics/article/4/2/249/245074)
+# 
+# **Alternative normalization methods**
+
+# In[2]:
+
+
+# Load arguments
+data_file = os.path.join(
+    os.path.dirname(os.getcwd()),
+    "data",
+    "input",
+    "Pa_compendium_02.22.2014.pcl")
+
+normalized_data_file = os.path.join(
+    os.path.dirname(os.getcwd()),
+    "data",
+    "input",
+    "train_set_normalized.pcl")
+
+metadata_file = os.path.join(
+    os.path.dirname(os.getcwd()),
+    "data",
+    "metadata",
+    "sample_annotations.tsv")
+
+
+# In[3]:
+
+
+# Read in data
+data = pd.read_table(data_file, header=0, sep='\t', index_col=0).T
+data.head(5)
+
+
+# In[4]:
+
+
+# Read in data
+normalized_data = pd.read_table(normalized_data_file, header=0, sep='\t', index_col=0).T
+normalized_data.head(5)
+
+
+# In[5]:
+
+
+# Read in metadata
+metadata = pd.read_table(metadata_file, header=0, sep='\t', index_col='ml_data_source')
```

Comment:
This column is used, because it is the one that appears to contain the sample id that corresponds to the gene expression data file.  But you're correct that I am loosing some samples since there are values missing.  I can also double check with Georgia about this.